packages/web/src/app/doc-pages/brand/color-doc/color-picker/color-picker-color-list/color-picker-color-list.component.ts

```diff
@@ -14,58 +16,38 @@ import { NeedsBorderPipe } from '../needs-border.pipe';
   imports: [NgClass, NeedsBorderPipe],
 })
 export class ColorPickerColorListComponent {
-  @Input({ required: true }) colorList: ColorsObject;
   @Input({ required: true }) category: 'primary' | 'secondary' | 'tertiary' | 'grey';
-  @Input({ required: true }) readonly theme: ThemeName;
-  @Output() chooseColor = new EventEmitter<ColorElement>();
-  @Output() userChoosesColor = new EventEmitter<void>();
-  @Input({ required: true })
-  get currentColor(): ColorElement {
-    return this._currentColor;
-  }
 
-  set currentColor(value: ColorElement | undefined) {
-    if (value) {
-      this._currentColor = value;
-    } else {
-      this._currentColor = {
-        ...this._currentColor,
+  private colorPickerService = inject(ColorPickerService);
 
-        //an empty color element, but keep the name for the empty state
-        contrast: { black: '', white: '' },
-        hex: '',
-        rgb: [-1, -1, -1],
-        tokens: [],
-      };
-    }
-  }
-  private _currentColor: ColorElement;
+  theme = this.colorPickerService.theme;
+  colorList = computed(() => (this.theme() === 'dark' ? darkColors : lightColors));
+  currentColor = this.colorPickerService.currentColor;
 
   chooseNewColor(color: ColorElement) {
-    this.chooseColor.emit(color);
-    this.userChoosesColor.emit();
+    this.colorPickerService.setCurrentColor(color.name);
   }
```

Comment:
Nit: if this method only uses `color.name`, maybe we can adjust the parameter to `colorName`?